atlp-rwanda/atlp-devpulse-fn
# #215 Ft tickets

### PR Description

This PR allows users to create tickets in case they face issues and get replies from the admin.

### Description of tasks that were expected to be completed

As an applicant I want to be able to open a ticket/inquiry about my application. This includes submitting a form with a title and a body explaining my inquiry. The tickets are sent to the admin (they can not be assigned to nor seen by another applicant).

If the applicant is not happy with the resolved, they can reply again which changes the ticket status back to applicant-reply

Note: Only the ticket author and admin can view tickets.

### How has this been tested?
- Clone repo
- Checkout branch `ft-tickets`
- Run `npm install`, then `npm run dev` to check if the app is running successfully
- Click on `tickets` on the sidebar to view available tickets
- If logged in as an admin, you will see all user tickets and if logged in as an applicant you will only see your tickets
- If logged in as an applicant, click `+ New Ticket`  button to create a new ticket
- On the actions, click view to see ticket information and reply to add a new reply to the ticket

### Credentials
username: `admin@example.com`
password: `password123`

username: `jean@example.com`
password: `password123`

### Track PR (issue number & link)
Issue: [#215](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/215)

### Screenshots (If appropriate especially on the frontend)
![Screenshot 2024-10-29 105008](https://github.com/user-attachments/assets/a247624d-eb2f-40fd-8bc0-1e4d37114a52)

![Screenshot 2024-10-29 105822](https://github.com/user-attachments/assets/f6c5147c-f4dc-42f3-9e13-e318dfc16245)

![Screenshot 2024-10-29 105313](https://github.com/user-attachments/assets/5448b5ed-90a3-4aa0-8845-5ec3ff11db25)

![Screenshot 2024-10-29 121944](https://github.com/user-attachments/assets/d6560f45-9e8a-4f0e-80b6-1c51951ebaca)
